minio/console
# Add Tiers improvements for Bucket Lifecycle management

There are places where we only need to list the Tiers' names including: Add Lifecycle Rule, Bulk add LifecycleRule and EditLifeCycleRule for Buckets. However we were using the `listTiers` API which does two extra requests including; getting tier stats and verification (verifies endpoint can perform needed requests).
Since we only need the names I've added a new api which does only that. 
There were also some issues in the tests for tiersList which I also addressed here.
This makes the requests much faster to give a better user experience.

### Changes:
- Add new API: `admin/tiers/names`
- Include tests for `getTiersName`
- Fix tests for `getTiers` (were were not checking the result from the response)
- Use of map to avoid iterating on the tierStats response on every tier for `getTiers`

### Test Steps:
With an authenticated users and TIERS configured do a `GET` on:
```
http://localhost:9090/api/v1/tiers/names
```
Response comes as:
```
{
    "items": [
        "PLAY-TIER",
        "SOME-TIER-CONFIG"
    ]
}
```

### Test Steps with UI:

For Bulk Lifecycle:
- Add Tiers
- Create two buckets with Versioning enabled
- Go to Administrator/Buckets and click on `Select Multiple buckets` Icon
- Select multiple buckets 
- Click on `Set Lifecycle` button
- Click on Transition (the list of tiers should appear)

For Add Bucket Lifecycle:
- Go to Administrator/Buckets and click on  one
- Click on Lifecycle
- Click on Add Lifecycle rule
- Click on Transition (the list of tiers should appear)

For Add Bucket Lifecycle:
- Go to Administrator/Buckets and click on  one with lifecycle rules
- Click on one rule
- List of tiers should appear

### Screenshots of it working:
Edit:
<img width="611" alt="Screenshot 2024-06-12 at 12 19 27 PM" src="https://github.com/minio/console/assets/11819101/154c8486-1f98-42e3-b1b4-eb8a0edb8710">
Add:
<img width="615" alt="Screenshot 2024-06-12 at 12 19 12 PM" src="https://github.com/minio/console/assets/11819101/10a9e606-70c0-4c08-a7a6-681b8229d5ad">
Bulk:
<img width="806" alt="Screenshot 2024-06-12 at 12 19 51 PM" src="https://github.com/minio/console/assets/11819101/e9dd6d6a-9528-4534-be51-9f1160f0b8e6">
